File: ls,cd/cli_ls_cd/simple-cli-app-master/src/main/java/ru.spbau.cliapp/core/TaskInfo.java
package ru.spbau.cliapp.core;

import java.util.Collections;
import java.util.List;

public class TaskInfo {
    public final String taskName;
    public final List<String> arguments;

    public TaskInfo(String taskName) {
        this(taskName, Collections.emptyList());
    }

    public TaskInfo(String taskName, List<String> arguments) {
        this.taskName = taskName;
        this.arguments = arguments;
    }
}

File: ls,cd/cli_lscd/simple-cli-app/src/main/kotlin/ru/spbau/cliapp/core/TaskInfo.kt
package ru.spbau.cliapp.core

/**
 * This class represents tasks for interpreter.
 *
 * Currently there's only two types of tasks:
 *
 * - [VarAssignmentInfo] represents assignment (like `x=10`)
 * - [TasksPipeline] represents pipeline of tasks described by [TaskInfo]s
 */
sealed class InterpreterTask

data class VarAssignmentInfo(val varName: String, val value: String) : InterpreterTask()
data class TasksPipeline(val tasks: List<TaskInfo>) : InterpreterTask()


/**
 * This class represents executable name and its arguments.
 */
data class TaskInfo(val taskName: String, val arguments: List<String> = emptyList())
